File: 1/1.py


def solve_a(data):
    big_carrier = max([sum(rations) for rations in data])
    return big_carrier


def solve_b(data):
    carriers_sorted = sorted([sum(rations) for rations in data])
    return sum(carriers_sorted[-3:])

if __name__ == "__main__":
    data = open('input.txt', 'r').read().split('\n\n')
    data = [rations.split('\n') for rations in data[:-1]]
    data = [[int(cals) for cals in rations] for rations in data]
    
    a = solve_a(data)

    b = solve_b(data)
    print(a, b)

File: 1/1.jl
f = readlines(open("input.txt"))

let
cals_list = []
cals = 0
for snack in f
    if isempty(snack)
        push!(cals_list, cals)
        cals = 0
    else
        cals += parse(Int64, snack)
    end
end

a = maximum(cals_list)
b = sum(sort(cals_list)[end-2:end])

println(a, " ", b)
end